submissions/homework_1/cc1759/hw.md

```diff
@@ -0,0 +1,123 @@
+- Date_Delivered: DATE
+
+
+I'm not sure how Amazon works for sellers, I would assume it just ties the products they sell to a regular account but im not sure. If so, it instead of list of sellers, this would just be a list of users, where the rows are user ID's and the columns are data such as the users name, users address, user payment info(idk if this would go here, that seems very insecure...), users listed products, and then other optional personalized info Amazon lets you put if you want like cloths size, pets, or intrests.
+
+
+For your list database I would go with a *Key-Value Database*, with each key being the name of the list and the value being a list of the product values ID's, which are then looked up and grabbed from the products table. Though one thing I will say, is that they are ordered by last placed in list and IDK if you can do that with Key_Value pairs but idk. Otherwise it could be a relational database with table being the list, row being the an auto incramented index, and the collums having the products ID. Then its also expandable to any other info you'd want to add like date added to the list or whatever.
+
+
+For recomendations, I think they would go with a *Vector Database* to search for similar products to what you've bought in the past, recently searched for, or have saved in your lists.
+
+
+Your cart would either be more simaler to the past orders database or the list database I'm not sure. I would think the lists database because I don't think it would need all the extra information like the past orders table, so maybe litterally just make it the same as the lists
+
+
+they probally also have a list for items ordered by everyone so they can ship the right things out to the right location.
+
+
+### **Discord**
+
+Discord probally has databases direct messages, for your friends, for your joined servers, server messages, and for users and there settings.
+
+
+For users it would probally have a *Relational Database*, with rows being your user ID (VARCHAR) and columns being things such as, disp_name, status, Tiene_Nitro, date joined, and all the settings (or they would have it)
+
+
+For severs joined, I would have to choose a *Graph Database*. Have each user and each server be a node, and have a user be connected to the servers they've joined by node. This should provide for easy lookup for mutual server with other users (though im not sure how the lookup would be any faster than a list?)
+
+
+For friend, I would have it be a *Graph Database* very simaler to the joined servers database, but with all the nodes being users, and the edges connecting those who are friends.
+
+
+For private messaging and server messages, it would probally be a JSON each JSON itself hold data for a server. In it are messages as objects with different properties (author, time posted, message contents) nested within which channel its in. Since the server/DM messaging and channels are probally in JSON format, I would us a *Document Database* 
```

Comment:
Typos:

```suggestion
For private messaging and server messages, it would probably be a JSON each JSON itself hold data for a server. In it are messages as objects with different properties (author, time posted, message contents) nested within which channel its in. Since the server/DM messaging and channels are probably in JSON format, I would us a *Document Database*
```